### PR TITLE
Add JOSS citation

### PR DESCRIPTION
**Pull request by josh146**

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  publisher:
+    name: Open Journals
+  start: 6720
+  title: "Catalyst: a Python JIT compiler for auto-differentiable hybrid
+    quantum programs"
+  type: article
+  url: "https://joss.theoj.org/papers/10.21105/joss.06720"
+  volume: 9
+title: "Catalyst: a Python JIT compiler for auto-differentiable hybrid
+  quantum programs"
```
Comment (rauletorresc) [inline CITATION.cff:66]: Maybe a new line at the end of the file?

**Review by rauletorresc — APPROVED** [review]

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,66 @@
+cff-version: "1.2.0"
+authors:
+- family-names: Ittah
+  given-names: David
+  orcid: "https://orcid.org/0000-0003-0975-6448"
+- family-names: Asadi
+  given-names: Ali
+- family-names: Lopez
+  given-names: Erick Ochoa
```
Comment (erick-xanadu) [inline CITATION.cff:9]:
```suggestion
- family-names: Ochoa Lopez
  given-names: Erick
```

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  Journal of Open Source Software.
+preferred-citation:
+  authors:
+  - family-names: Ittah
+    given-names: David
+    orcid: "https://orcid.org/0000-0003-0975-6448"
+  - family-names: Asadi
+    given-names: Ali
+  - family-names: Lopez
+    given-names: Erick Ochoa
```
Comment (erick-xanadu) [inline CITATION.cff:38]:
```suggestion
  - family-names: Ochoa Lopez
    given-names: Erick
```

**Review by erick-xanadu — CHANGES_REQUESTED** [review]
Just two-last-names problems :)

**Comment by dime10** [comment]
Just had a look at the rendered version, not sure if due to this PR but this (`quantum classical programs.`) looks out of place:
Screenshot 2024-07-09 at 5 36 13 PM

File: CITATION.cff (hunk 2 above)
Comment (josh146) [reply CITATION.cff:9]: Thanks @erick-xanadu :) This file was automatically generated by JOSS and I didn't proofread it carefully!

File: CITATION.cff (hunk 1 above)
Comment (josh146) [reply CITATION.cff:66]:
```suggestion
  quantum programs"

```

File: doc/index.rst [hunk 4]
```diff
@@ -53,8 +53,12 @@ Catalyst
    <br>
 
 .. mdinclude:: ../README.md
-  :start-line: 19
-  :end-line: 72
+  :start-line: 20
+  :end-line: 73
+
+.. mdinclude:: ../README.md
+  :start-line: 143
```
Comment (josh146) [inline doc/index.rst:60]:
```suggestion
  :start-line: 142
```

**Comment by josh146** [comment]
@dime10 thanks for catching that -- it should now be fixed

**Review by erick-xanadu — APPROVED** [review]

**Pull request merged**